### ISSUE TITLE
What's the difference between the three methods of epoch or iteration?

### ISSUE THREAD

**Issue by HZJ6** [JappaB/Active_Learning_Object_Detection#3]
parser.add_argument('--train_epochs', default=10, type=int,
                    help= 'Number of epochs')
parser.add_argument('--fixed_number_of_epochs', default=None, type=int,
                    help= 'Fill a fixed number of train iterations to use that instead of early stopping based on a validation set')
parser.add_argument('--fixed_number_of_train_iterations', default=None, type=int,
                    help= 'Train for a fixed number of iterations, like in the localization stability paper')

**Comment by JappaB** [comment]
Hi there @HZJ6 ,
My apologies for the extremely late reply. I've been focussed on other things for quite some time. I'm not sure what each parameter does in the code that I put here eventually (as mentioned, it's research code, not production code, so quite a bit of spaghetti code duct-taped together :))

However, one of the problems I faced was: when do you decide to stop with training when comparing several iterations of the active learning algorithm. [Early stopping](https://www.google.com/search?q=early+stopping+machine+learning&tbm=isch&source=iu&ictx=1&vet=1&fir=sghIGOpmXhdPYM%252CUNl7q3l9FrJFvM%252C%252Fm%252F01f78j%253B7FrfSFwcXAgyBM%252Cyn5FVCtjJquD4M%252C_%253BwBlQ-V7rstN92M%252COWuB24MyvCmGqM%252C_%253BhUD0IV4_C18lAM%252COWuB24MyvCmGqM%252C_&usg=AI4_-kQ5ScLtmiT5HOPQML80sAyo7H0mfA&sa=X&ved=2ahUKEwik3ZLE1Pf1AhXWt6QKHUm6AKQQ_B16BAgsEAE#imgrc=sghIGOpmXhdPYM) in ML theoretically should give you the best result for each epoch. On the other hand, what if you are just (un)lucky what the validation set of a certain experiment. If a validation set is particularly easy, this will result on an artificially high score, which doesn't necessarily generalise well.
In theory it shouldn't matter, you do enough runs of the same experiment, and you'll converge on which active learning method works better. My experiment runs where however quite computationally expensive, so in the end I wanted to reduce the number of parameters that influenced the result of the experiment (having a good or a bad validation set was one).
